bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairBaseBridgeHandler.java

```diff
@@ -46,23 +62,42 @@
  * @author Jeff James - Initial contribution
  *
  */
-public abstract class PentairBaseBridgeHandler extends BaseBridgeHandler {
+@NonNullByDefault
+public abstract class PentairBaseBridgeHandler extends BaseBridgeHandler implements CallbackPentairParser {
     private final Logger logger = LoggerFactory.getLogger(PentairBaseBridgeHandler.class);
 
     /** input stream - subclass needs to assign in connect function */
-    protected BufferedInputStream reader;
-    /** output stream - subclass needs to assing in connect function */
-    protected BufferedOutputStream writer;
+    protected Optional<BufferedInputStream> reader = Optional.empty();
+    /** output stream - subclass needs to assign in connect function */
+    protected Optional<BufferedOutputStream> writer = Optional.empty();
+
+    protected PentairParser parser = new PentairParser();
```

Comment:
Please make any fields that can be final, final.
```suggestion
    protected final PentairParser parser = new PentairParser();
```